### ISSUE TITLE
[iOS] Compilation error if used with google_maps_flutter (>=2.0.0) package

### ISSUE THREAD

**Issue by sperochon** [derTuca/flutter_google_places_picker#35]
Hello,

I use
```
google_places_picker: ^2.1.0+3
google_maps_flutter: ^2.0.1
```

and compiled on iOS.

I've got this cocoapod error:
```CocoaPods could not find compatible versions for pod "GoogleMaps":
      In Podfile:
        google_maps_flutter (from `.symlinks/plugins/google_maps_flutter/ios`) was resolved to 0.0.1, which depends on
          GoogleMaps (< 3.10)

        google_places_picker (from `.symlinks/plugins/google_places_picker/ios`) was resolved to 0.0.3, which depends on
          GoogleMaps (~> 4.0)

    There are only pre-release versions available satisfying the following requirements:

    	'GoogleMaps', '< 3.10'

    	'GoogleMaps', '~> 4.0'

    You should explicitly specify the version in order to install a pre-release version
```

I've then forked this project and removed the use of GoogleMaps lib in .podspec file since it seems it is not used at all except
`[GMSServices provideAPIKey:apiKey];`
(that I have removed too).

I have also change
`s.dependency 'GooglePlaces', '~> 4.0'`
to
`s.dependency 'GooglePlaces', '~> 3.0'`
to make it compiling fine.

Everything is now compiling and working fine.

So, why is the GoogleMaps lib needed on iOS?
And why is the GooglePlaces lib version upgrade to 4? Shouldn't be a minimum of 3 since official flutter google_maps package uses this version?

Thanks!

**Comment by sperochon** [comment]
Ok, maybe I'm all wrong... please do not work on this issue right now, I'm looking at it again... sorry ;-)

**Comment by derTuca** [comment]
In all honesty, the Google Maps dependency is probably a relic from when the Place Picker was still a thing, so we really should look into removing it. From what I can see, the Places library now has its own way of providing the API key.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by sperochon** [comment]
Thanks @derTuca for your quick answer

After more investigation, it appears that there is also another part to make it compiling and working fine:
https://github.com/flutter/flutter/issues/65102#issuecomment-806100386

**Comment by derTuca** [comment]
I've just removed GoogleMaps from the plugin podfile.

**Comment by derTuca** [comment]
@sperochon can you try recompiling it now (without the fix) please? I'd like to know so I know if I can close the issue or not. Thanks!

**Comment by sperochon** [comment]
@derTuca I've look at your code modifications and seems ok for me. So you can close this issue.

FYI, I can not test it inside my project since it needs Google Maps 3.x and your plugin "depends on
          GooglePlaces (~> 4.0) was resolved to 4.0.0, which depends on
            GoogleMaps/Base (= 4.0.0)"

See https://github.com/flutter/flutter/issues/65102#issuecomment-806100386

Thanks again.
